engine/dlib/src/dmsdk/dlib/configfile_gen.hpp

```diff
@@ -0,0 +1,66 @@
+// Generated, do not edit!
+// Generated with cwd=/Users/mathiaswesterdahl/work/defold/engine/dlib and cmd=/Users/mathiaswesterdahl/work/defold/scripts/dmsdk/gen_sdk.py -i /Users/mathiaswesterdahl/work/defold/engine/dlib/sdk_gen.json
+
+// Copyright 2020-2024 The Defold Foundation
+// Copyright 2014-2020 King
+// Copyright 2009-2014 Ragnar Svensson, Christian Murray
+// Licensed under the Defold License version 1.0 (the "License"); you may not use
+// this file except in compliance with the License.
+// 
+// You may obtain a copy of the License, together with FAQs at
+// https://www.defold.com/license
+// 
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#ifndef DMSDK_CONFIGFILE_GEN_HPP
+#define DMSDK_CONFIGFILE_GEN_HPP
+
+#if !defined(__cplusplus)
+   #error "This file is supported in C++ only!"
+#endif
+
+
+/*# Configfile_gen
+ *
+ * description
+ *
+ * @document
+ * @name Configfile_gen
+ * @namespace dmConfigFile
+ * @path dmsdk/dlib/configfile_gen.hpp
+ */
+
+#include <stdint.h>
+
+#include <dmsdk/dlib/configfile.h>
+
+namespace dmConfigFile
+{
+    /*#
+    * Generated from [ref:HConfigFile]
+    */
+    typedef HConfigFile HConfig;
```

Comment:
Parsing the documentation using clang doesn't really give a good representation (e.g. false positives)
Also, it's C documentation, so to automatically mangle it to support C# won't really be feasible.
So, a reference back to the C header is our first step.